base_layer/core/src/base_node/sync/block_sync/synchronizer.rs

```diff
@@ -238,22 +239,6 @@ impl<B: BlockchainBackend + 'static> BlockSynchronizer<B> {
                 block.accumulated_data.accumulated_monero_difficulty,
                 block.accumulated_data.accumulated_blake_difficulty,
             );
-            current_block = Some(block);
-        }
-
-        if let Some(block) = current_block {
-            // Update metadata to last tip header
-            let header = &block.block.header;
-            let height = header.height;
-            let best_block = header.hash();
-            let accumulated_difficulty = block.accumulated_data.total_accumulated_difficulty;
-            self.db
-                .write_transaction()
-                .set_best_block(height, best_block.to_vec(), accumulated_difficulty)
-                .commit()
-                .await?;
-
-            self.hooks.call_on_complete_hooks(block);
```

Comment:
`call_on_complete_hooks` is used by block sync base node state to notify the mempool that the chain has changed. This needs to be called after the loop with the current tip block.

Reply:
yeah but it calls it twice? or am I reading the code wrong?
Its called in line 227

Reply:
227 is different, that is the "progress" hook  
```rust
  self.hooks
                .call_on_progress_block_hooks(block.clone(), tip_height, &[peer.clone()]);
```